dashboard/app/controllers/pd/mimeo_sso_controller.rb

```diff
@@ -0,0 +1,76 @@
+# Authenticate and redirect a Code Studio user to Mimeo using their single-sign-on (SSO) API
+class Pd::MimeoSsoController < ApplicationController
+  load_resource :enrollment, class: 'Pd::Enrollment', find_by: :code, id_param: :enrollment_code
+
+  ENDPOINT = 'https://my.mimeo.com/sso/authenticate.ashx'.freeze
+  AUTHORIZED_MARKETPLACE_URL = 'unpluggedlessonmaterials'.freeze
+  SSO_VERSION = '2.0'.freeze
+
+  ALLOWED_COURSES = [Pd::Workshop::COURSE_CSF].freeze
+
+  def authenticate_and_redirect
+    # Make sure the user is logged in and is the owner of this enrollment
+    authorize! :manage, @enrollment
+    raise CanCan::AccessDenied unless ALLOWED_COURSES.include? @enrollment.workshop.course
+
+    unless @enrollment.completed_survey?
+      # Until the survey is processed, it won't show up in the enrollment model.
+      # Check pegasus forms directly before giving up.
+      raise CanCan::AccessDenied unless PEGASUS_DB[:forms].where(kind: 'PdWorkshopSurvey', source_id: @enrollment.id).any?
+    end
+
+    # Set up fields for the Mimeo SSO API. Each of these will be rendered as a hidden field on a form which will be
+    # automatically posted to the @endpoint on page load.
+    # Mimeo's API will then authenticate and either redirect the user's browser to the redirectUrl on success,
+    # or to our errorUrl if it fails (see #error below).
+    @endpoint = ENDPOINT
+    @fields = {
```

Comment:
In their [example C# code](https://github.com/Mimeo/Connect/blob/master/SSO/RSA%20Examples/Mimeo-SSO.CS), instead of javascript auto-submitting, there is a button the user has to press in order to submit the form and start the process. It's designed such that these form fields may be visible. I chose to auto-submit to remove the extra step of pushing a button. I also saw examples of doing it this way online. Pausing is not really a big deal. 

Most of the fields are encrypted with Mimeo's RSA public key. Only they have the private key to decrypt. The encrypted fields are `firstName`, `lastName`, `email`, `organizationId`, and `companyName`. Strangely their API expects `companyId` to be unencrypted.

So it's true a savvy user could scrape and reuse the fields, including the encrypted ones. However if they reuse their own name and email, they will simply log back into the same account. And in order to spoof and create a new account, the will need to have the RSA public key (which is secret) to encrypt a new name and email that is valid and can be decrypted and understood on the other end.